### ISSUE TITLE
[QUESTION] Does ffmpeg use hardware acceleration on Linux

### ISSUE THREAD

**Issue by M4TH76** [iteufel/nwjs-ffmpeg-prebuilt#39]
Hi 
I'm using nwjs on Arch Linux to play h264 videos and It doesn't seem like the decoder use the GPU (intel I5 6gen)
I've checked with intel_gpu_top and the GPU doesn't go over 6-7% and CPU is 80-90%
Can nwjs-ffmpeg-prebuilt use HW acceleration?
Thanks for the help

**Comment by iteufel** [comment]
@M4TH76 
Try adding `--ignore-gpu-blacklist` to your [chromium-args](http://docs.nwjs.io/en/latest/References/Manifest%20Format/#chromium-args)
Open `chrome://gpu/` with nw.js and check if your GPU is blacklisted.
You can also try Installing the [mesa](https://www.archlinux.org/packages/extra/x86_64/mesa/) and  [libva-intel-driver](https://www.archlinux.org/packages/extra/x86_64/libva-intel-driver/) packages.

https://wiki.archlinux.org/index.php/Hardware_video_acceleration

**Comment by M4TH76** [comment]
I've already added the arguments to chromium 

Vainfo list h264 as supported

chrome://media-internals list video_decoder:ffmpeg... Instead of gpu.. 
That's why a asked if ffmpeg can use the gpu to decode video like it does on Windows

**Comment by Ajeey** [comment]
@M4TH76 Any chance you enabled hardware acceleration on MAC? If yes, whats the approach you used?

**Comment by M4TH76** [comment]
@Ajeey sorry not a Mac user...

Bump...
Can nwjs-ffmpeg-prebuilt use HW acceleration?

**Comment by mocudev** [comment]
From what I understand, you actually need to apply a patch to chromium in order to enable H.264 hardware accelerated decoding. I played around with it a little bit a while back and managed to produce a version of nwjs with H.264 hardware decode enabled.

I put up a repo that has the patches for nwjs 15 (I uploaded a binary version of nwjs also): https://github.com/mocudev/nwjs_linux_hwdecode

There's also a branch where I started trying to applying a newer patch to nwjs 25, however I had some issues getting it working and haven't had time to keep working on it. This is where the patches are that I started going off of for the nwjs 25 branch: https://chromium-review.googlesource.com/c/chromium/src/+/532294
